File: src/main/java/com/choru/lock/domain/StockFacade.java
package com.choru.lock.domain;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class StockFacade {
    private final StockService service;

    public void decrease(Long id, Long quantity) throws InterruptedException{
        while (true){
            try{
                service.decrease(id, quantity);

                break;
            }catch (Exception e){
                Thread.sleep(50);
            }
        }
    }

}

File: src/main/kotlin/com/choru/lock/domain/StockFacade.kt
package com.choru.lock.domain

import org.springframework.stereotype.Component

@Component
class StockFacade(
    private val service: StockService
) {
    @Throws(InterruptedException::class)
    fun decrease(id: Long, quantity: Long) {
        while (true) {
            try {
                service.decrease(id, quantity)
                break
            } catch (e: Exception) {
                Thread.sleep(50)
            }
        }
    }
}
